src/widgets/index/__tests__/index-test.ts

```diff
@@ -434,6 +442,13 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index/js/"
             distinct: true,
           })
         );
+
+        // `instantSearchInstance` must have been notified 4 times of the `uiState` changes:
+        // 1. By the widget, for the widget initialization
+        // 2. By the helper `change` event callback, for the change to the query parameters
+        // 3. By the widget, for the children widget disposal
+        // 4. By the helper `change` event callback, also for the children widget disposal
```

Comment:
Is it expected? Not critical since the call to onChange is deferred, but maybe the call to `onChange` in `removeWidgets` is unnecessary.

WDYT?

Reply:
Indeed I think the it's useless, the call below (`setState`) trigger the `change` event which already contains `onChange`. You may have noticed that we compute two times the `uiState` on remove. Once before `setState` to be able to correctly get the `SearchParameters`, once after when the event is emitted (regular flow). We might be able to avoid the two consecutive calls but at the moment it's good enough.